### PR TITLE
RFC: add method to ImageStack to use Napari for show stack

### PR DESCRIPTION
**Pull request by kevinyamauchi**
As discussed in #677, the nascent [Napari](https://github.com/Napari) viewer can view large stacks and has performant scrolling between slices. Also, it automatically generates the appropriate number of sliders for multidimensional stacks. It seems useful for debugging and optimizing starfish pipeines, so I've added an adapter I've added some notes below. Please let me know your thoughts.

* Since the viewer isn't fully featured, I've created a separate method (`show_stack_napari()`), so we can continue to use all current viewers.
* `show_stack_napari()` uses the get_slices method to create the data arrays to pass to Napari, so we can slice the ImageStack. 
* Since scrolling through all dimensions in a full 5D ImageStack is not a possibility, we may want to add an easy option to just pass the whole stack (e.g., `indices='all'`) to Napari rather than giving all indices. 
* I used the instructions in REQUIREMENTS.txt to update it - please let me know if I did that correctly.

### CODE REVIEW

**Comment by kevinyamauchi** [comment]
P.S. I didn't encode the test notebook as a py file because I think we'd nuke it before merging.

**Comment by ttung** [comment]
How big are these packages?  I imagine that PyQt is probably ginormous.

One possible solution is to move the import into `show_stack_napari()`, wrap it like:

```
try:
    import napari_gui
except ImportError:
    warnings.warn("Cannot find the napari library. Install it by running \"pip install napari\"")
    return

<rest of show_stack_napari()>
```

**Comment by kevinyamauchi** [comment]
Good point, @ttung. Pardon my ignorance, but in the case that napari_gui is only imported with that method, does napari_gui go in the `REQUIREMENTS.txt`?

**Comment by ttung** [comment]
I don't think we have a good solution for this problem (of bloating requirements) yet.  I would prefer not to block you while we sort things out.  If @ambrosejcarr and @dganguli approve this, I have no problem having this land and we can fix the requirements problem afterwards.

File: starfish/imagestack/imagestack.py [hunk 1]
```diff
@@ -386,6 +387,28 @@ def set_slice(
 
         self._data.values[slice_list] = data
 
+    def show_stack_napari(self, indices: Mapping[Indices, Union[int, slice]]):
```
Comment (ambrosejcarr) [inline starfish/imagestack/imagestack.py]: This is awesome. Is it possible to have napari label the sliders? I find that part of it a bit unintuitive. 

If it's _not_ possible, we should document the order that they appear on the GUI from top to bottom. Based on the MERFISH example, seems like it's the same as the `reordered_array`?

File: starfish/imagestack/imagestack.py [hunk 2]
```diff
@@ -386,6 +387,28 @@ def set_slice(
...
+            Indices to select a volume to visualize. Passed to `Image.get_slice()`.
+            See `Image.get_slice()` for examples.
+
+        Notes
+        -----
+        To use in a Jupyter notebook, use the %gui qt5 magic.
+
+
+        """
+        # Switch axes such that it is indexed [x, y, rnd, c, z]
```
Comment (ambrosejcarr) [inline starfish/imagestack/imagestack.py]: I'd prefer `[x, y, r, c, z]` or `[x, y, round, channel, z]`

**Pull request merged**